homeassistant/components/openuv/__init__.py

```diff
@@ -127,53 +118,54 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     _verify_domain_control = verify_domain_control(hass, DOMAIN)
 
     websession = aiohttp_client.async_get_clientsession(hass)
-    openuv = OpenUV(
-        hass,
-        entry,
-        Client(
-            entry.data[CONF_API_KEY],
-            entry.data.get(CONF_LATITUDE, hass.config.latitude),
-            entry.data.get(CONF_LONGITUDE, hass.config.longitude),
-            altitude=entry.data.get(CONF_ELEVATION, hass.config.elevation),
-            session=websession,
-        ),
+    client = Client(
+        entry.data[CONF_API_KEY],
+        entry.data.get(CONF_LATITUDE, hass.config.latitude),
+        entry.data.get(CONF_LONGITUDE, hass.config.longitude),
+        altitude=entry.data.get(CONF_ELEVATION, hass.config.elevation),
+        session=websession,
     )
 
-    # We disable the client's request retry abilities here to avoid a lengthy (and
-    # blocking) startup:
-    openuv.client.disable_request_retries()
+    async def async_update_protection_data() -> dict[str, Any]:
+        """Update binary sensor (protection window) data."""
+        low = entry.options.get(CONF_FROM_WINDOW, DEFAULT_FROM_WINDOW)
+        high = entry.options.get(CONF_TO_WINDOW, DEFAULT_TO_WINDOW)
+        return await client.uv_protection_window(low=low, high=high)
 
-    try:
-        await openuv.async_update()
-    except HomeAssistantError as err:
-        LOGGER.error("Config entry failed: %s", err)
-        raise ConfigEntryNotReady from err
+    coordinators: dict[str, OpenUvCoordinator] = {
+        coordinator_name: OpenUvCoordinator(
+            hass,
+            name=coordinator_name,
+            latitude=client.latitude,
+            longitude=client.longitude,
+            update_method=update_method,
+        )
+        for coordinator_name, update_method in (
+            (DATA_UV, client.uv_index),
+            (DATA_PROTECTION_WINDOW, async_update_protection_data),
+        )
+    }
 
-    # Once we've successfully authenticated, we re-enable client request retries:
-    openuv.client.enable_request_retries()
+    async def async_update_data() -> None:
+        """Update all data from OpenUV."""
+        init_tasks = [
+            coordinator.async_config_entry_first_refresh()
```

Comment:
It looks like you are calling this outside of `async_setup_entry` from `update_data`.  Since `async_config_entry_first_refresh` can raise `ConfigEntryNotReady`, you probably need another version that uses `async_refresh` instead to avoid getting an unexpected exception if something goes wrong with the update

Reply:
Good catch. Will resolve tomorrow.

Reply:
Actually had a moment now. Thanks!